### PR TITLE
Intelligent Autocompletions

### PR DESCRIPTION
**Pull request by curran**
Builds out the foundations for 

 * #147 

Done in an epic pair programming session with @EvWhymark , based on previous work found in https://github.com/vizhub-core/vzcode/pull/294 and https://github.com/vizhub-core/vzcode/pull/295.

### CODE REVIEW

File: src/client/useTypeScript/index.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+// }    ,[content])
+
+// // // Updates the TypeScript worker with the current content
+// // // of the files.
+// // typeScriptWorker.postMessage({
+// //     event: 'update-content',
+// //     details: content,
+// //   });
+
+// };
```
Comment (EvWhymark) [inline src/client/useTypeScript/index.ts:86]:
```suggestion
```
Should probably remove unnecessary comment.

File: src/client/useTypeScript/index.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
+//     import { JSONOp } from 'ot-json1';
+// import {
+//   FileId,
+//   ShareDBDoc,
+//   VZCodeContent,
+// } from '../../types';
+// import { useEffect, useRef, useState,useCallback } from 'react';
```
Comment (EvWhymark) [inline src/client/useTypeScript/index.ts:7]:
```suggestion
```
Unnecessary comment

**Review by EvWhymark — APPROVED** [review]
Just some leftover comments from our thinking that we should probably remove but otherwise looks good to me! Really glad we were able to finish this up and get it working.

**Comment by curran** [comment]
It "works" but the completions are generic, something is wrong.

**Comment by curran** [comment]
Got it!

image

**Pull request merged**